### PR TITLE
Adding UIA accessibility support for PictureBox control

### LINKED ISSUE #2164: Narrator focus is not moving in proper logical order when user is navigating by using caps + arrow keys (tab index is ignored)
* .NET Core Version: 3.1
* Have you experienced this same bug with .NET Framework?: Yes

**Problem description:**
Narrator focus is not moving in proper logical order when user is navigating by using caps + arrow keys: Narrator goes thru the controls in order which is the order of controls being added to the Controls collection of the form in design time (order of controls can be corrected by manual changing the order of adding the control to the collection of child controls.

**Actual behavior:**
image

**Expected behavior:**
Narrator caps + arrow navigation should take into account control tab index and accessible hierarchy should also take into account control tab index. 

**Minimal repro:**
* Create WinForms Application with one form
* Add TextBox1 to the bottom part of the form
* Add TextBox2 to the top part of form
* Add TextBox3 to the middle part of form
* Set tab index of TextBox2 to 1
* Set tab index of TextBox3 to 2
* Set tab index of TextBox1 to 3
* Run the app and navigate using Narrator.
* Observe Narrator caps + arrow right navigation goes firstly to TextBox1 -> TextBox2 -> TextBox3, but should go accordingly to tab index TextBox2 -> TextBox3 -> TextBox1

### LINKED ISSUE #3232: PictureBox control does not support UIA accessibility
* .NET Core Version: 

.NET Core 5.0
Version: 5.0.100-alpha.1.20073.10
Commit: 29f4d693a9
Runtime Environment:
OS Name: Windows
OS Version: 10.0.18363
OS Platform: Windows
RID: win10-x64
Base Path: C:\Program Files\dotnet\sdk\5.0.100-alpha1-05536
Host (useful for support):
Version: 5.0.0-alpha.1.20072.3
Commit: c3dc1fdfdc

* Have you experienced this same bug with .NET Framework?: 

Yes

**Problem description:**

image

Accessibility providers:
[pid:23288,providerId:0x70F0E Main:Nested [pid:23476,providerId:0x70F0E Main(parent link):Microsoft: MSAA Proxy (unmanaged:UIAutomationCore.dll)]; Hwnd(parent link):Microsoft: HWND Proxy (unmanaged:uiautomationcore.dll)]"

**Expected behavior:**
There should be UIA accessibility provider from WinForms.

**Minimal repro:**
Create WinForms app with PictureBox control and analyze it using Inspect.

Related item:
This work item is the sub of #2164

### PR DESCRIPTION
**Pull request by M-Lipin**
Fixes #3232 

## Proposed changes

- Adding UIA accessibility support for PictureBox control.
- Moving PictureBox accessible object implementation to a separate file.

## Customer Impact

- Improving development experience for ListView control accessibility.
- Prerequisite for #2164.

## Regression? 

- No

## Risk

- Minimal

### Before

image

### After

image

## Test methodology 

- Manual testing;
- Unit tests (to be implemented);
- UI automation.

## Accessibility testing  

- Inspect;
- Accessibility Insights;
- Narrator and other accessibility client apps.
 

## Test environment(s) 

.NET Core 5.0
Version: 5.0.100-alpha.1.20073.10
Commit: 29f4d693a9
Runtime Environment:
OS Name: Windows
OS Version: 10.0.18363
OS Platform: Windows
RID: win10-x64
Base Path: C:\Program Files\dotnet\sdk\5.0.100-alpha1-05536
Host (useful for support):
Version: 5.0.0-alpha.1.20072.3
Commit: c3dc1fdfdc

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/winforms/pull/3233)

### CODE REVIEW

File: src/System.Windows.Forms/src/System/Windows/Forms/PictureBox.PictureBoxAccessibleObject.cs [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+#nullable disable
+
+using System.Runtime.InteropServices;
+using static Interop;
+
+namespace System.Windows.Forms
+{
+    public partial class PictureBox
+    {
+        [ComVisible(true)]
```
Comment (weltkante) [inline src/System.Windows.Forms/src/System/Windows/Forms/PictureBox.PictureBoxAccessibleObject.cs:14]: don't add new ComVisible attributes

File: src/System.Windows.Forms/src/System/Windows/Forms/PictureBox.PictureBoxAccessibleObject.cs [hunk 2]
```diff
@@ -0,0 +1,38 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable disable
```
Comment (RussKie) [inline src/System.Windows.Forms/src/System/Windows/Forms/PictureBox.PictureBoxAccessibleObject.cs:5]: We want NRT

File: src/System.Windows.Forms/src/System/Windows/Forms/PictureBox.PictureBoxAccessibleObject.cs (hunk 1 above)
Comment (SergeySmirnov-Akvelon) [reply src/System.Windows.Forms/src/System/Windows/Forms/PictureBox.PictureBoxAccessibleObject.cs]: Thanks, fixed.

File: src/System.Windows.Forms/src/System/Windows/Forms/PictureBox.PictureBoxAccessibleObject.cs (hunk 2 above)
Comment (SergeySmirnov-Akvelon) [reply src/System.Windows.Forms/src/System/Windows/Forms/PictureBox.PictureBoxAccessibleObject.cs]: Thanks, enabled NRT.

File: src/System.Windows.Forms/src/System/Windows/Forms/PictureBox.PictureBoxAccessibleObject.cs [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+                    _ => base.GetPropertyValue(propertyID)
+                };
+
+            internal override bool IsPatternSupported(UiaCore.UIA patternId)
+                => patternId switch
+                {
+                var p when
+                    p == UiaCore.UIA.LegacyIAccessiblePatternId => true,
+                _ => base.IsPatternSupported(patternId)
+    };
```
Comment (RussKie) [inline src/System.Windows.Forms/src/System/Windows/Forms/PictureBox.PictureBoxAccessibleObject.cs:38]: incorrect indentation

File: src/System.Windows.Forms/src/System/Windows/Forms/PictureBox.PictureBoxAccessibleObject.cs [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+                    UiaCore.UIA.ControlTypePropertyId
+                        => UiaCore.UIA.PaneControlTypeId,
+                    UiaCore.UIA.AutomationIdPropertyId
+                        => Owner.Name,
+                    UiaCore.UIA.IsKeyboardFocusablePropertyId
+                        => true,
+                    _ => base.GetPropertyValue(propertyID)
+                };
+
+            internal override bool IsPatternSupported(UiaCore.UIA patternId)
```
Comment (vladimir-krestov) [inline src/System.Windows.Forms/src/System/Windows/Forms/PictureBox.PictureBoxAccessibleObject.cs:32]: You can remove this due to #3412

File: src/System.Windows.Forms/tests/UnitTests/AccessibleObjects/PictureBox.PictureBoxAccessibleObjectTests.cs [hunk 5]
```diff
@@ -0,0 +1,82 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Xunit;
+using static Interop.UiaCore;
+
+namespace System.Windows.Forms.Tests
+{
+    public class PictureBox_PictureBoxAccessibleObjectTests
```
Comment (vladimir-krestov) [inline src/System.Windows.Forms/tests/UnitTests/AccessibleObjects/PictureBox.PictureBoxAccessibleObjectTests.cs:10]: Please add `IsHandleCreated` checking for all your tests

File: src/System.Windows.Forms/tests/UnitTests/AccessibleObjects/PictureBox.PictureBoxAccessibleObjectTests.cs [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+        {
+            using var pictureBox = new PictureBox();
+            var pictureBoxAccessibleObject = new PictureBox.PictureBoxAccessibleObject(pictureBox);
+
+            Assert.NotNull(pictureBoxAccessibleObject.Owner);
+            Assert.Equal(AccessibleRole.Client, pictureBoxAccessibleObject.Role);
+        }
+
+        [WinFormsFact]
+        public void PictureBoxAccessibleObject_Property_returns_correct_value()
```
Comment (vladimir-krestov) [inline src/System.Windows.Forms/tests/UnitTests/AccessibleObjects/PictureBox.PictureBoxAccessibleObjectTests.cs:29]: Please divide this to 3 tests

File: src/System.Windows.Forms/tests/UnitTests/AccessibleObjects/PictureBox.PictureBoxAccessibleObjectTests.cs [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+            Assert.True(pictureBox.IsHandleCreated);
+            object value = pictureBoxAccessibleObject.GetPropertyValue(propertyID);
+            Assert.Equal(expected, value);
+
+            // TODO: ControlAccessibleObject shouldn't force handle creation, tracked in https://github.com/dotnet/winforms/issues/3062
+            Assert.True(pictureBox.IsHandleCreated);
+        }
+
+        [WinFormsFact]
+        internal void PictureBoxAccessibleObject_IsPatternSupported_returns_correct_value()
```
Comment (vladimir-krestov) [inline src/System.Windows.Forms/tests/UnitTests/AccessibleObjects/PictureBox.PictureBoxAccessibleObjectTests.cs:70]: Rework the test name

File: src/System.Windows.Forms/src/System/Windows/Forms/PictureBox.PictureBoxAccessibleObject.cs [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+
+using static Interop;
+
+namespace System.Windows.Forms
+{
+    public partial class PictureBox
+    {
+        internal class PictureBoxAccessibleObject : ControlAccessibleObject
+        {
+            internal PictureBoxAccessibleObject(PictureBox owner) : base(owner)
```
Comment (RussKie) [inline src/System.Windows.Forms/src/System/Windows/Forms/PictureBox.PictureBoxAccessibleObject.cs:13]:
```suggestion
            public PictureBoxAccessibleObject(PictureBox owner) : base(owner)
```

**Comment by vladimir-krestov** [comment]
Testers approved the fix ✔️

**Pull request merged**